src/commands/aksInspektorGadget/aksInspektorGadget.ts

```diff
@@ -0,0 +1,173 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { getKubernetesClusterInfo, KubernetesClusterInfo } from '../utils/clusters';
+import { getExtensionPath, longRunning } from '../utils/host';
+import { Errorable, failed } from '../utils/errorable';
+import * as tmpfile from '../utils/tempfile';
+import { getKubectlGadgetBinaryPath } from '../utils/helper/kubectlGadgetDownload';
+import { invokeKubectlCommand } from '../utils/kubectl';
+import path = require('path');
+import { createWebView, getRenderedContent, getResourceUri } from '../utils/webviews';
+
+enum InspektorGadget {
+    Deploy,
+    Undeploy
+}
+
+export async function aksInspektorGadgetDeploy(
+    _context: IActionContext,
+    target: any
+): Promise<void> {
+    await gadgetDeployUndeploy(_context, target, InspektorGadget.Deploy)
+}
+
+export async function aksInspektorGadgetUnDeploy(
+    _context: IActionContext,
+    target: any
+): Promise<void> {
+    await gadgetDeployUndeploy(_context, target, InspektorGadget.Undeploy);
+}
+
+async function gadgetDeployUndeploy(
+    _context: IActionContext,
+    target: any,
+    gadget: InspektorGadget
+): Promise<void> {
+    const kubectl = await k8s.extension.kubectl.v1;
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+
+
+    if (!kubectl.available) {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return undefined;
+    }
+
+    if (!cloudExplorer.available) {
+        vscode.window.showWarningMessage(`Cloud explorer is unavailable.`);
+        return undefined;
+    }
+
+    if (!clusterExplorer.available) {
+        vscode.window.showWarningMessage(`Cluster explorer is unavailable.`);
+        return undefined;
+    }
+
+    const clusterInfo = await getKubernetesClusterInfo(target, cloudExplorer, clusterExplorer);
+    if (failed(clusterInfo)) {
+        vscode.window.showErrorMessage(clusterInfo.error);
+        return undefined;
+    }
+
+    await prepareInspektorGadgetInstall(clusterInfo.result, gadget, clusterInfo.result.kubeconfigYaml, kubectl);
+}
+
+async function prepareInspektorGadgetInstall(
+    cloudTarget: KubernetesClusterInfo,
+    gadget: InspektorGadget,
+    kubeconfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>
+): Promise<void> {
+    const clustername = cloudTarget.name;
+
+    switch (gadget) {
+        case InspektorGadget.Deploy:
+            await deployGadget(clustername, kubeconfig, kubectl);
+            return;
+        case InspektorGadget.Undeploy:
+            const answer = await vscode.window.showInformationMessage(`Do you want to undeploy gadget in selected cluster?`, "Yes", "No");
```

Comment:
Would be good to specify the cluster name: `Do you want to undeploy gadget in ${clustername}?`